app/src/context/auth/AuthContextController.tsx

```diff
@@ -29,10 +27,29 @@ export const AuthContextController = ({ children }: AuthContextControllerProps)
       setHasActiveSession(true);
     }
 
-    supabase.auth.onAuthStateChange((_event, s) => {
+    const { data: authListener } = supabase.auth.onAuthStateChange(async (event, s) => {
       setSession(s);
       setHasActiveSession(true);
+
+      if (event === "SIGNED_IN" && s?.user?.id) {
+        try {
+          await fetch("/api/auth", {
+            method: "POST",
+            headers: new Headers({ "Content-Type": "application/json" }),
+            credentials: "same-origin",
+            body: JSON.stringify({ event, session: s }),
+          });
+
+          await createBusinessMutation();
```

Comment:
This mutation is actually not needed here. It's for testing the integration between the Server Side supabase queries and the client. 

We need to create a business record once the user enters their dashboard and clicks on "apply". 
Actually, the business record should be created by BSats admins only (for now)

Reply:
Actually, I'm thinking that the actual name of the table should be: `campaign`, because the user may have many businesses, or only one, but what we create in BSat are funding campaigns. 

So the actual mutation would be: `createBusinessCampaign`, or simply: `createCampaign`, or `createFundingCampaign`

Reply:
Agree